kyaml/yaml/rnode.go

```diff
@@ -248,6 +443,391 @@ func (rn *RNode) IsNilOrEmpty() bool {
+// passed in, and the value will be stored in ptr. ptr can be any concrete type
+// (e.g. string, []corev1.Container, []string, corev1.Pod, map[string]string) or
+// a RNode. RNode should be used if you are dealing with comments that is more
+// than what LineComment, HeadComment, SetLineComment and SetHeadComment can
+// handle. It returns if the field is found and a potential error.
+func (rn *RNode) Get(ptr interface{}, fields ...string) (bool, error) {
+	found, err := func() (bool, error) {
+		if rn.IsNil() {
+			return false, ErrRNodeNotFound{}
+		}
+
+		if ptr == nil || reflect.ValueOf(ptr).Kind() != reflect.Ptr {
+			return false, fmt.Errorf("ptr must be a pointer to an object")
+		}
+
+		if rnptr, ok := ptr.(*RNode); ok {
+			val, found, err := rn.nestedValue(fields...)
+			if err != nil || !found {
+				return found, err
+			}
+			rnptr.SetYNode(val.value)
+			return found, err
+		}
+
+		switch k := reflect.TypeOf(ptr).Elem().Kind(); k {
+		case reflect.Struct, reflect.Map:
+			val, found, err := rn.nestedValue(fields...)
+			if err != nil || !found {
+				return found, err
+			}
+			if val.value.Kind != yaml.MappingNode {
+				return found, fmt.Errorf("unable to get field %v, since the RNode was not a MappingNode", fields)
+			}
+			// We are not using val.value.Decode(ptr) here, since the k8s core types don't
+			// have yaml tags.
+			err = val.As(ptr)
+			return found, err
+		case reflect.Slice:
+			val, found, err := rn.nestedValue(fields...)
+			if err != nil || !found {
+				return found, err
+			}
+			if val.value.Kind != yaml.SequenceNode {
+				return found, fmt.Errorf("unable to get field %v, since the RNode was not a SequenceNode", fields)
+			}
+			// We are not using val.value.Decode(ptr) here, since the k8s core types don't
+			// have yaml tags.
+			err = val.As(ptr)
+			return found, err
+		case reflect.String:
+			s, found, err := rn.GetNestedString(fields...)
+			if err != nil || !found {
+				return found, err
+			}
+			*(ptr.(*string)) = s
+			return found, nil
+		case reflect.Int, reflect.Int64:
+			i, found, err := rn.GetNestedInt(fields...)
+			if err != nil || !found {
+				return found, err
+			}
+			if k == reflect.Int {
+				*(ptr.(*int)) = i
+			} else if k == reflect.Int64 {
+				*(ptr.(*int64)) = int64(i)
+			}
+			return found, nil
+		case reflect.Float64:
+			f, found, err := rn.GetNestedFloat(fields...)
+			if err != nil || !found {
+				return found, err
+			}
+			*(ptr.(*float64)) = f
+			return found, nil
+		case reflect.Bool:
+			b, found, err := rn.GetNestedBool(fields...)
+			if err != nil || !found {
+				return found, err
+			}
+			*(ptr.(*bool)) = b
+			return found, nil
+		default:
+			return false, fmt.Errorf("unhandled kind %s", k)
+		}
+	}()
+	if err != nil {
+		return found, fmt.Errorf("unable to get fields %v as %T with error: %w", fields, ptr, err)
+	}
+	return found, nil
+}
+
+func (rn *RNode) nestedValue(fields ...string) (*RNode, bool, error) {
```

Comment:
I noticed on the one hand that the majority of the callers are doing ` err != nil || !found`, and on the other that we already have a type `NoFieldError`. Should we return that instead of the third value?